app/controllers/topics_controller.rb

```diff
@@ -1,12 +1,12 @@
 class TopicsController < ApplicationController
   def index
     @topics = Topic.all
-    render json: @topics, status: 200
+    render json: TopicSerializer.new(@topics).serializable_hash, status: 200
   end
 
   def show
     @topic = Topic.find(params[:id])
-    render json: @topic, status: 201
+    render json: TopicSerializer.new(@topic).serializable_hash, status: 201
```

Comment:
Since this is a `show` action, I believe `201` is an incorrect status. According to the [json spec ](https://jsonapi.org/format/#fetching-resources-responses-200) `:ok` should be the correct status for a single resource search.

Note: That was a mistake on my part in the last commit but since your working on the json spec, is a good opportunity for you to address it.